### PR TITLE
reset selected filters after search

### LINKED ISSUE #1693: Active filter must reset after i search for something else
**Is your feature request related to a usage problem (not a bug)? Please describe.**
After u enabled a filter and searched for something, the filter stays enabled when u search for something else.

**Describe the solution you'd like to see implemented**
See video.

**Describe alternatives you've considered**
None.

**Video**

https://user-images.githubusercontent.com/73130443/133058050-c1525e31-cea9-4438-a3f5-00d0d17ea3b6.mp4

https://user-images.githubusercontent.com/73130443/133058057-a5f759fc-5171-4381-aaaf-f662a9fae857.mp4

**Additional context**
Edit: the thing that i didnt show but also must be implemented is when u just go to another page lets say most popular the filter also clears.

### PR DESCRIPTION
**Pull request by ankitchauhan-aka**
## Pull Request Type

- [ ] Bugfix
- [x] Feature Implementation
- [ ] Documentation
- [ ] Other

## Related issue
closes #1693 

## Testing
1. Apply filters, You must have a red border that shows filter(s) are selected
2. Then search for a term in the search box and choose any option from autosuggestion dropdown
3. After selecting autosuggested options, Filters must be reset to default options.
4. The red border from the filter icon will be removed

### CODE REVIEW

**Review by PikachuEXE — CHANGES_REQUESTED** [review]
I tested with these steps and failed
- Enter something, enter search result via enter/search button
- Change filter options
- Press search button again
- Search result not updated but filter reset

image

**Comment by ankitchauhan-aka** [comment]
> I tested with these steps and failed
> 
> * Enter something, enter search result via enter/search button
> * Change filter options
> * Press search button again
> * Search result not updated but filter reset
> 
> image

@PikachuEXE I've just attached a test video with mentioned steps and not able to replicate issue. Could you please share similar so I can fix it asap.

**Comment by PikachuEXE** [comment]
https://github.com/FreeTubeApp/FreeTube/assets/1018543/0f64ab14-8c03-4538-92a8-835ab0072b00

**Review by efb4f5ff-1298-471a-8973-3d47447115dc — CHANGES_REQUESTED** [review]
Filter not applied when:

1. Application starts
2. apply labels
3. enter query
4. hit search


https://github.com/FreeTubeApp/FreeTube/assets/73130443/6e017af0-8814-41b8-bcdf-3426fdd8709b

**Comment by ankitchauhan-aka** [comment]
> Filter not applied when:
> 
> 1. Application starts
> 2. apply labels
> 3. enter query
> 4. hit search
> 
>  VirtualBoxVM_E7hDKoPbOO.mp4
**Fixed.**

Thanks for the review.

**Comment by PikachuEXE** [comment]
It reset filters under strange conditions (e.g. like changing filter below

https://github.com/FreeTubeApp/FreeTube/assets/1018543/b29fa63c-bf96-4fd6-b632-6c5be57e189e

Probably like
- Search A
- Enter B
- Try to set multiple criteria

**Comment by efb4f5ff-1298-471a-8973-3d47447115dc** [comment]
@PikachuEXE this is not a bug but intended behavior. There is how YT also behaves. In your example Time Today resets because you clicked on Type Channel. Time is based on upload date. A channel cant have a upload date.

**Review by PikachuEXE — APPROVED** [review]
OK there are several issue discovered during testing but not introduced by this PR

**Comment by efb4f5ff-1298-471a-8973-3d47447115dc** [comment]
Issue?

**Review by efb4f5ff-1298-471a-8973-3d47447115dc — APPROVED** [review]

**Comment by ankitchauhan-aka** [comment]
@jasonhenriquez can we get this reviewed.

File: src/renderer/components/ft-search-filters/ft-search-filters.vue [hunk 1]
```diff
@@ -44,6 +47,7 @@
           :title="$t('Search Filters.Duration.Duration')"
           :labels="durationLabels"
           :values="durationValues"
+          :selected="searchSettings.duration"
```
Comment (jasonhenriquez) [inline src/renderer/components/ft-search-filters/ft-search-filters.vue:50]: **suggestion (blocking):** You should be able to remove the `ft-radio-button` and `ft-search-filters` changes altogether once you have implemented the other suggested change.

**Review by jasonhenriquez — CHANGES_REQUESTED** [review]
Looks like the implementation was changed enough by #3975 to make the required alteration easier. Address review comments, re-validate, and you should be good.

**Comment by jasonhenriquez** [comment]
Hi @ankitchauhan-aka, are you still available to make these final alterations?

**Comment by ankitchauhan-aka** [comment]
Hi @jasonhenriquez 
Yeah. I was too much busy in my job will do it asap

**Comment by ankitchauhan-aka** [comment]
@jasonhenriquez seems good now!

**Comment by jasonhenriquez** [comment]
In my testing, it's not actually applying the filters now - the probable issue is that the removal is processing before the results are being grabbed.

**Comment by ankitchauhan-aka** [comment]
I think that's correct behaviour. It should remove filter if search query is modified.

**Comment by jasonhenriquez** [comment]
No, I mean it's not applying the filter even the first time.

**Comment by ankitchauhan-aka** [comment]
https://www.awesomescreenshot.com/video/28282864?key=3f195c5137a0c69969519b2bdf7fe124
@jasonhenriquez Please take a look at the attached video, if you feel that some of the records do not match the time filter, So it's not related to our application logic, Even YouTube has the same results. or if your testing produces different results than mine please share a capture.

**Comment by ankitchauhan-aka** [comment]
@jasonhenriquez could you please share!

**Comment by jasonhenriquez** [comment]
Hi @ankitchauhan-aka, sorry for the delay. Just retested now; it was working the first handful of times, then just stopped applying the filter at all.

https://github.com/FreeTubeApp/FreeTube/assets/84899178/4d5c08d3-63a4-43ef-845f-c85deb5db1ab

Before moving forward with this, I'd also like some discussion on the intended behavior. We're showing the filter as active after a search is made, until the user submits another query. That's not intuitive, as we shouldn't be showing the filter as if it's active if it isn't going to be. The main reason YT has this behavior the way it is is because changing a filter setting re-applies the same search but with the filter changed. If we do want this feature, the filters should show as clear immediately after the search processes.

**Comment by ankitchauhan-aka** [comment]
@jasonhenriquez thanks for the video, I can see the issue in your video but am not able to reproduce it. Did you notice any steps for the issue? Also, I've modified it for the new requirement to clear the filter after search

**Review by jasonhenriquez — APPROVED** [review]
LGTM. The bug that I was encountering was setting a feature filter while a channel was set, which should be fixed by #5197.

**Comment by ankitchauhan-aka** [comment]
Great can we merge it now?

**Comment by ankitchauhan-aka** [comment]
@jasonhenriquez great. Hoping to get it merged before we get any conflicts again 😉

**Review by efb4f5ff-1298-471a-8973-3d47447115dc — CHANGES_REQUESTED** [review]
> when u just go to another page lets say most popular the filter also clears.

Taken from the linked issue. This also needs to be implemented.

**Comment by ankitchauhan-aka** [comment]
> > when u just go to another page lets say most popular the filter also clears.
> 
> Taken from the linked issue. This also needs to be implemented.

Fixed.

**Review by efb4f5ff-1298-471a-8973-3d47447115dc — CHANGES_REQUESTED** [review]
Route is changed, Filters applied correctly but Filter icon isnt active

https://github.com/FreeTubeApp/FreeTube/assets/73130443/b5a0f26a-189d-4d1a-b76f-ba573061fb37

**Comment by jasonhenriquez** [comment]
@efb4f5ff-1298-471a-8973-3d47447115dc Isn't that the expected behavior? See [my earlier comment](https://github.com/FreeTubeApp/FreeTube/pull/5033#issuecomment-2153371087):

Before moving forward with this, I'd also like some discussion on the intended behavior. We're showing the filter as active after a search is made, until the user submits another query. That's not intuitive, as we shouldn't be showing the filter as if it's active if it isn't going to be. The main reason YT has this behavior the way it is is because changing a filter setting automatically re-submits the same search query but with the new filter. If we do want this feature, the filters should show as clear immediately after the search processes.

**Comment by efb4f5ff-1298-471a-8973-3d47447115dc** [comment]
Oof i missed that. Will retest soon.

**Comment by efb4f5ff-1298-471a-8973-3d47447115dc** [comment]
@jasonhenriquez isnt this intuitive? Now the user needs to remember what filter they applied if they want to refine their search more:

1. Apply filter
2. Input search query
3. Hit search
4. Filters reset
5. User has to apply the same filter(s) again + they like to add to refine their search results.

User has to remember what filters they applied in their previous search. That seems like a headache to me.

https://github.com/FreeTubeApp/FreeTube/assets/73130443/f9e13178-69ba-463b-9acf-6145baaf1d36

**Comment by ankitchauhan-aka** [comment]
@jasonhenriquez so what's next step about this?

**Comment by ankitchauhan-aka** [comment]
I think we should keep the filters selected until the user is changing search query.

**Comment by jasonhenriquez** [comment]
If I'm understanding correctly, that means showing it as active for the next search, even if it won't be, unless the user then changes the filter to something else, in which case it will be? That seems a bit confusing if so.

**Comment by ankitchauhan-aka** [comment]
What if clear filter only if user types different search query

**Comment by jasonhenriquez** [comment]
If you can get the filter to show as non-active to the user unless they re-open the search filter modal, I'm fine with that. I would also be fine with "have the filters be forgotten unless the user opts to open the modal after a search." I just get the sense that we're trying to emulate YT's behavior here somewhat, which doesn't work exactly, since our filter button is omnipresent, and our current behavior is different. 

I'm also wondering if maybe this is getting too complicated / potentially confusing to users, and maybe we should just have a one-click button that removes all active search filters instead.

**Pull request closed without merging**